spec/models/va_profile_redis/contact_information_v2_spec.rb

```diff
@@ -0,0 +1,336 @@
+      end
+    end
+  end
+
+  describe '#response' do
+    context 'when the cache is empty' do
+      it 'caches and return the response', :aggregate_failures do
+        allow_any_instance_of(
+          VAProfile::V2::ContactInformation::Service
+        ).to receive(:get_person).and_return(person_response)
+
+        VCR.use_cassette('va_profile/v2/contact_information/person', VCR::MATCH_EVERYTHING) do
+          if VAProfile::Configuration::SETTINGS.contact_information.cache_enabled
+            expect(contact_info.redis_namespace).to receive(:set).once
+          end
+          expect_any_instance_of(VAProfile::V2::ContactInformation::Service).to receive(:get_person).twice
+          expect(contact_info.status).to eq 200
+          expect(contact_info.response.person).to have_deep_attributes(person)
+        end
+      end
+    end
+
+    context 'when there is cached data' do
+      it 'returns the cached data', :aggregate_failures do
+        VCR.use_cassette('va_profile/v2/contact_information/person', VCR::MATCH_EVERYTHING) do
+          contact_info.cache(user.uuid, person_response)
+          expect_any_instance_of(VAProfile::V2::ContactInformation::Service).not_to receive(:get_person)
+          expect(contact_info.response.person).to have_deep_attributes(person)
+        end
+      end
+    end
+  end
+
+  describe 'contact information attributes' do
+    context 'with a successful response' do
+      before do
+        allow(VAProfile::Models::V2::Person).to receive(:build_from).and_return(person)
+        allow_any_instance_of(
+          VAProfile::V2::ContactInformation::Service
+        ).to receive(:get_person).and_return(person_response)
+      end
+
+      describe '#email' do
+        it 'returns the users email address object', :aggregate_failures do
+          VCR.use_cassette('va_profile/v2/contact_information/person', VCR::MATCH_EVERYTHING) do
+            expect(contact_info.email).to eq person.emails.first
+            expect(contact_info.email.class).to eq VAProfile::Models::Email
+          end
+        end
+      end
+
+      describe '#residential_address' do
+        it 'returns the users residential address object', :aggregate_failures do
+          residence = address_for VAProfile::Models::V2::Address::RESIDENCE
+          VCR.use_cassette('va_profile/v2/contact_information/person', VCR::MATCH_EVERYTHING) do
+            expect(contact_info.residential_address).to eq residence
+            expect(contact_info.residential_address.class).to eq VAProfile::Models::V2::Address
+          end
+        end
+      end
+
+      describe '#mailing_address' do
+        it 'returns the users mailing address object', :aggregate_failures do
+          residence = address_for VAProfile::Models::V2::Address::CORRESPONDENCE
+          VCR.use_cassette('va_profile/v2/contact_information/person', VCR::MATCH_EVERYTHING) do
+            expect(contact_info.mailing_address).to eq residence
```

Comment:
Maybe call this `correspondence` instead? 
```suggestion
          correspondence = address_for VAProfile::Models::V2::Address::CORRESPONDENCE
          VCR.use_cassette('va_profile/v2/contact_information/person', VCR::MATCH_EVERYTHING) do
            expect(contact_info.mailing_address).to eq correspondence
```

Reply:
This is a completely difference address and I would have to re-record the specs.